zulip_bots/zulip_bots/lib.py

```diff
@@ -68,7 +68,28 @@ def show_error_and_exit(self) -> None:
         sys.exit(1)
 
 
-class StateHandler:
+class BotIdentity:
+    def __init__(self, name: str, email: str) -> None:
+        self.name = name
+        self.email = email
+        self.mention = '@**' + name + '**'
+
+class BaseStorage(Protocol):
+    def put(self, key: Text, value: Any) -> None:
+        ...
+
+    def get(self, key: Text) -> Any:
+        ...
+
+    def contains(self, key: Text) -> bool:
+        ...
+
+class BotStorage(BaseStorage):
```

Comment:
I think We can Include the methods of `BaseStorage` in `BotStorage`, am not sure why we need a separate class for `BaseStorage`

Reply:
Because the return value of `use` needs to be BaseStorage. If it's in `BotStorage`, I can't annotate it to return a `BotStorage` as it's not yet defined.